### PR TITLE
Disable client cache in teleterm local proxy integration tests

### PR DESCRIPTION
**Pull request by ravicious**
This PR aims to address flakiness of lib/teleterm integration tests (#39268, https://github.com/gravitational/teleport/issues/20906#issuecomment-1992137366).

I realize that this is not the best solution as we still don't understand what the exact problem is (#39268 includes attempts at figuring this out). However, we are not able to reproduce the issue locally and the client cache (#15603) is the only recent big change that I can think of which could cause flakiness.

Included in this PR are also fixes to logging and removing an unnecessary client.Close(). Best reviewed commit-by-commit.

### CODE REVIEW

File: lib/teleterm/services/clientcache/clientcache.go [hunk 1]
```diff
@@ -155,14 +153,10 @@ func (c *Cache) Clear() error {
...
 
-func (c *Cache) addToCache(clusterURI uri.ResourceURI, proxyClient *client.ProxyClient) error {
+func (c *Cache) addToCache(clusterURI uri.ResourceURI, proxyClient *client.ProxyClient) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	var err error
-	if c.clients[clusterURI] != nil {
-		err = c.clients[clusterURI].Close()
-	}
```
Comment (gzdunek) [inline lib/teleterm/services/clientcache/clientcache.go:163]: Right, this wasn't needed.

**Review by gzdunek — APPROVED** [review]
Looks good, let's see if this helps.

File: integration/proxy/teleterm_test.go [hunk 2]
```diff
@@ -197,6 +198,7 @@ func testGatewayCertRenewal(ctx context.Context, t *testing.T, params gatewayCer
 		CreateTshdEventsClientCredsFunc: func() (grpc.DialOption, error) {
 			return grpc.WithTransportCredentials(insecure.NewCredentials()), nil
 		},
+		ClientCache:    clientcache.NewNoCache(storage),
```
Comment (gzdunek) [inline integration/proxy/teleterm_test.go:201]: Hmm, do we want to leave a comment why `no cache` is used?

**Review by zmb3 — APPROVED** [review]

File: integration/proxy/teleterm_test.go (hunk 2 above)
Comment (zmb3) [reply integration/proxy/teleterm_test.go:201]: You guys are welcome to do this in a follow up PR, but I had to merge this as it was killing the team's productivity and ability to get a v15 release out.

**Pull request merged**